source/slang/slang-emit.cpp

```diff
@@ -214,6 +215,66 @@ static void dumpIRIfEnabled(
     }
 }
 
+static void reportCheckpointIntermediates(CodeGenContext* codeGenContext, DiagnosticSink* sink, IRModule* irModule)
+{
+    // Report checkpointing information
+    EndToEndCompileRequest *compileRequest = codeGenContext->isEndToEndCompile();
```

Comment:
why do we need this? We should get the optionset direclty from the `codeGenConetxt->targetProgram` instead of the end to end request.